docs/user-guide/framework-spec.md

```diff
@@ -156,32 +233,41 @@ Command: TEXT | REQUIRED
 
 Vendors: TEXT_LIST | IN_LIST(pre-fined list of all cloud vendors)
 
-### Construct Level Attributes
-
-#### Shared attributes
+#### Construct Level Attributes
 
 Name: SIMPLE_TEXT | REQUIRED
 Description: MARKDOWN_TEXT
 Icon: URL
 CloudVendors: TEXT_LIST | IN_LIST(pre-fined list of all cloud vendors)
 DefaultCloudVendor: TEXT | IN_LIST(pre-fined list of all cloud vendors)
 
-#### Network
+#### Types Definition via Plan JSON
```

Comment:
I'm finding this section quite difficult to follow. I don't see the connect with the rest and how to put it together. Seems like a package author needs to understand how to create this.

Reply:
if not needed lets drop. presume was an oversight that you left in? I'm not missing something right?

Reply:
you reply to the general comment now contradicts my understanding.

Reply:
so do we need the types in a JSON file or not?

Reply:
We do